Add the Themes and Styles topic; remove applyTheme mentions (#5548)
surveyjs/survey-library: docs/manage-default-themes-and-styles.md

# Themes & Styles

This help topic describes how to add SurveyJS themes to your Angular, Vue, React, Knockout, or jQuery application and switch between them if you want to apply multiple themes.

## Add SurveyJS Themes to Your Application

> SurveyJS ships with the Modern and Default V2 UI themes illustrated below.

Author: "V2" was added to distinguish the new version of the Default theme from the old one. The old one is now obsolete, but the name stuck. As a rule, we don't enclose names in quotes because in most cases capitalization is enough.

> SurveyJS ships with the Modern and Default V2 UI themes illustrated below.

Reviewer: OK, I see your point. Thank you for the clarification.

[View Demo](https://surveyjs.io/form-library/examples/survey-customcss/ (linkStyle))

In addition, the `SurveyModel` object raises events that allow you to override CSS classes for individual questions, panels, pages, and choice items. Refer to the following event descriptions in the API reference for more information:

- [`onUpdateQuestionCssClasses`](https://surveyjs.io/form-library/documentation/api-reference/survey-data-model#onUpdateQuestionCssClasses)
- [`onUpdatePanelCssClasses`](https://surveyjs.io/form-library/documentation/api-reference/survey-data-model#onUpdatePanelCssClasses)
- [`onUpdatePageCssClasses`](https://surveyjs.io/form-library/documentation/api-reference/survey-data-model#onUpdatePageCssClasses)
- [`onUpdateChoiceItemCss`](https://surveyjs.io/form-library/documentation/api-reference/survey-data-model#onUpdateChoiceItemCss)

[View Demo](https://surveyjs.io/form-library/examples/survey-cssclasses/ (linkStyle))

## Switch Between Themes

> If you add more than one SurveyJS theme to your application, call the `applyTheme(themeName)` method to specify the active theme. Pass one of the following theme names as the method's argument:

Reviewer 2: ```suggestion
If you add more than one SurveyJS theme to your application, call the `applyTheme(themeName)` method to specify an active theme. Pass one of the following theme names as the method's argument:
```

- `"defaultV2"`
- `"modern"`

The following theme names are also supported, but they are obsolete:

- `"default"`
- `"orange"`
- `"darkblue"`
- `"darkrose"`
- `"stone"`
- `"winter"`
- `"winterstone"`

The code example below shows how to apply the Default V2 theme:

```js
import { StylesManager } from 'survey-core';

StylesManager.applyTheme("defaultV2");
```

> Previously, you needed to call the `applyTheme(themeName)` method even if your application includes only one SurveyJS theme. Since v1.9.69, a theme applies automatically if it is the only SurveyJS theme in the application.

## Bootstrap Support (Obsolete)

SurveyJS includes themes designed for use with Bootstrap and Bootstrap Material. However, they are considered obsolete and no longer receive updates.

> Previously, Bootstrap integration was part of the main package. Since v1.9.69, it is shipped as a separate module/script. The following code shows how to apply the Bootstrap or Bootstrap Material theme by importing a module:

Author: ```suggestion
Previously, Bootstrap integrations were part of the main package. Since v1.9.69, they are shipped as separate modules/scripts. The following code shows how to apply the Bootstrap or Bootstrap Material theme by importing a module:
```

It is a module when it is added to a modular application, and a script when it is added to a monolithic application. You can see code examples for these two cases below.

> Previously, Bootstrap integration was part of the main package. Since v1.9.69, it is shipped as a separate module/script. The following code shows how to apply the Bootstrap or Bootstrap Material theme by importing a module:

Reviewer: Thank you for the clarification. And I like your new version.

```js
// Apply the Bootstrap theme
import { bootstrapThemeName } from "survey-core/plugins/bootstrap-integration";
import { StylesManager } from 'survey-core';

StylesManager.applyTheme(bootstrapThemeName);
```

```js
// Apply the Bootstrap Material theme
import { bootstrapMaterialThemeName } from "survey-core/plugins/bootstrap-material-integration";
import { StylesManager } from 'survey-core';

StylesManager.applyTheme(bootstrapMaterialThemeName);
```

The code below shows how to apply the themes by referencing a script:

```html
<!-- Apply the Bootstrap theme -->
<script src="https://unpkg.com/survey-core/plugins/bootstrap-integration.min.js"></script>
```
```js
Survey.StylesManager.applyTheme("bootstrap");
```

```html
<!-- Apply the Bootstrap Material theme -->
<script src="https://unpkg.com/survey-core/plugins/bootstrap-material-integration.min.js"></script>
```
```js
Survey.StylesManager.applyTheme("bootstrapmaterial");
```

> Bootstrap theme customization in code has also changed. Objects that specify applied CSS classes have been moved to the separate module/script.

Author: The definite article was supposed to convey that the object were moved to the same integration module/script we talked about previously.

```suggestion
Bootstrap theme customization in code has also changed. Objects that specify applied CSS classes have been moved to the integration modules/scripts mentioned above.
```

> Bootstrap theme customization in code has also changed. Objects that specify applied CSS classes have been moved to the separate module/script.

Reviewer: Looks good to me. Thank you.
